src/i_system.c

```diff
@@ -205,13 +206,56 @@ void I_PrintStartupBanner(const char *gamedescription)
 boolean I_ConsoleStdout(void)
 {
 #ifdef _WIN32
-    // SDL "helpfully" always redirects stdout to a file.
-    return false;
+    return _isatty(_fileno(stdout));
 #else
     return isatty(fileno(stdout));
 #endif
 }
 
+#ifdef _WIN32
+static void ReopenOutputHandle(DWORD std, FILE *stream)
+{
+    HANDLE handle = GetStdHandle(std);
+    DWORD lpmode;
+
+    if (GetConsoleMode(handle, &lpmode))
+    {
+        freopen("CONOUT$", "w", stream);
+    }
+}
+
+void I_WinConsole(void)
+{
+    char console_env[2];
+
+    // Console wrapper will set _console=1 if we were called using the com
+    // file.
+    if (GetEnvironmentVariable("_console", console_env, 2) &&
```

Comment:
You used `SetEnvironmentVariableW` with the `wchar_t` parameter, but here you read in the buffer `char console_env[2]`. I assume it was converted behind the scene, but still for the sake of consistency I think we should use `wchar_t` here too.

Reply:
I argue in this case it's not a big deal because the value being stored is the character `1` and that should always be available in any ANSI code page. However you raise a good point in general about reading environment variables on Windows, especially those like `DOOMWADDIR` or `DOOMWADPATH` which store paths. Elsewhere Chocolate Doom uses `getenv` which, on Windows, will always return ANSI representations based on the current code page. This is a recipe for trouble since the file I/O functions assume strings are UTF8.

I think there probably needs to be wrapper functions for getting and setting environment variables. They would convert between UTF16LE and UTF8 on Windows, similar to what you implemented for files. That'd be a project for another PR though.

Reply:
Yes, you're right, it's not necessary to use the Unicode Win API functions, it's just my old habit. Windows internal representation is in UTF-16, but since we're building Chocolate Doom without Unicode support, we're getting all strings in the local OEM code page, which is ANSI compliant, so there's no need to convert `getenv` either. All file handling functions should already be covered.

Reply:
Going down the rabbit hole on this...

Something I learned just recently is that if you use SDL2 main redirection (which I believe Chocolate is doing), it will provide `argv` strings in UTF-8, not the local code page. See: https://github.com/libsdl-org/SDL/blob/cdf9010a19512dc84f6770d3617b848f5848876f/src/main/windows/SDL_windows_main.c#L38-L91

This is why I think we need to convert environment variable strings (well, at least ones that store paths) to UTF-8 as well, so that they match those in the `argv` array.

Also, given the above, I think this call in w_file_win32.c might be wrong: https://github.com/chocolate-doom/chocolate-doom/blob/67715d6e2725322e6132e9ff99b9a2a3f3b10c83/src/w_file_win32.c#L97-L99

Maybe this is causing the issue that @JNechaevsky noticed with trying to load a non-latin named wad?

Reply:
Interesting find, I didn't know. But since UTF-8 is ANSI compatible, it doesn't matter to Choco, since all `argv` manipulations are done in ANSI space.

In this function `MultiByteToWideChar(CP_OEMCP...` string converted from UTF-8 to CP_OEMCP, so it still works fine. Although Microsoft recommends using UTF-8 or UTF-16 here: https://docs.microsoft.com/en-us/windows/win32/api/stringapiset/nf-stringapiset-multibytetowidechar Also this line maybe problematic:
https://github.com/chocolate-doom/chocolate-doom/blob/67715d6e2725322e6132e9ff99b9a2a3f3b10c83/src/w_file_win32.c#L92 
`MAX_PATH` on Windows is only 260 characters, this is for backwards compatibility only.

I can't reproduce @JNechaevsky's issue on Windows 10 and Windows XP in a virtual machine, even with Total Commander. Could it be a problem with Windows 11?